pkg/rego/schemas/builder.go

```diff
@@ -0,0 +1,270 @@
+package schemas
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/aquasecurity/defsec/pkg/state"
+	"github.com/aquasecurity/trivy-policies/pkg/rego/convert"
+)
+
+type RawSchema struct {
+	Type       string               `json:"type"` // object
+	Properties map[string]Property  `json:"properties,omitempty"`
+	Defs       map[string]*Property `json:"definitions,omitempty"`
+}
+
+type Property struct {
+	Type       string              `json:"type,omitempty"`
+	Ref        string              `json:"$ref,omitempty"`
+	Properties map[string]Property `json:"properties,omitempty"`
+	Items      *Property           `json:"items,omitempty"`
+}
+
+type builder struct {
+	schema RawSchema
+}
+
+func Build() (*RawSchema, error) {
+
+	b := newBuilder()
+
+	inputValue := reflect.ValueOf(state.State{})
```

Comment:
@nikpivkin `state.State{}` comes from defsec today and is shared across trivy-iac and trivy-aws as well. It's one of those dependencies which is hard to get away from as it's shared between all of the components.